File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        csum=nums[0]
        psum=nums[0]
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1]:
                csum+=nums[i]
            else:
                psum=max(csum,psum)
                csum=nums[i]
        return(max(csum,psum))
        

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int ctotal=nums[0];
        int ptotal=nums[0];
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i-1]<nums[i])
            {
                ctotal+=nums[i];
            }
            else 
            {
                ptotal=Math.max(ptotal,ctotal);
                ctotal=nums[i];
            }
        }
        return Math.max(ctotal,ptotal); 
    }
}